homeassistant/components/ihc/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow for IHC integration."""
+import logging
+
+from ihcsdk.ihccontroller import IHCController
+import voluptuous as vol
+
+from homeassistant import config_entries, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import AbortFlow, FlowResult
+
+from .const import CONF_AUTOSETUP, DOMAIN
+from .util import get_controller_serial
+
+CONFIG_FLOW_VERSION = 1
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL, default="http://192.168.1.3"): str,
+        vol.Required(CONF_USERNAME, default=""): str,
+        vol.Required(CONF_PASSWORD, default=""): str,
+        vol.Optional(CONF_AUTOSETUP, default=True): bool,
+    }
+)
+
+
+def dovalidate(hass: HomeAssistant, user_input) -> str:
```

Comment:
```suggestion
def do_validate(hass: HomeAssistant, user_input) -> str:
```

Reply:
dovalidate has been renamed to do_validate